### ISSUE TITLE
Tests not loaded for new Angular projects from ng new 

### ISSUE THREAD

**Issue by speedyscooter** [lucono/karma-test-explorer#38]
**Describe the Bug**

running ng new and pointing vscode to folder no tests show.  Also any debugging levels are ignored.

**Steps to Reproduce the Behavior**

1. ng new app
2. add routing
3. css
4. open vscode to folder.
**Describe the Expected Behavior**

- expected tests to show up..

**Provide Debug Logs**

- With the `karmaTestExplorer.logLevel` set to `debug`, please capture and attach the logs from the `Karma Test Explorer` Output chanel when the issue occurs.
- Please also capture and attach the logs from the `Karma Server` Output channel when the issue occurs.
```
[2022-07-26 19:02:04.424] [DEBUG] [Main]: Reloading workspace 'C:/work/angular/pdm' due to changed settings: ["projects"]
[2022-07-26 19:02:04.424] [DEBUG] [ProjectFactory]: Inspecting workspace folder for testing: c:\work\angular\pdm
[2022-07-26 19:02:04.425] [DEBUG] [ProjectFactory]: Excluding projects in workspace folder 'C:/work/angular/pdm' - 'karmaTestExplorer.enableExtension' is set to false
[2022-07-26 19:02:04.425] [DEBUG] [ProjectFactory]: Excluding projects in workspace folder - None of the inclusion conditions were satisfied: C:/work/angular/pdm
```
**Screenshots**

- If applicable, add screenshots to help explain the issue you are having.

**Please Provide the Following Information**
 
 - OS: Windows
 - Environment: local install
 - Node Version: (e.g. 14.x)
 - VS Code Version: 
 - Version: 1.69.2 (user setup)
Commit: 3b889b090b5ad5793f524b5d1d39fda662b96a2a
Date: 2022-07-18T16:12:52.460Z
Electron: 18.3.5
Chromium: 100.0.4896.160
Node.js: 18.6.0
V8: 10.0.139.17-electron.0
OS: Windows_NT x64 10.0.19044
 - Karma Version:6.4.0
 - Frameworks: jasmine

**Additional Context**
```json
{
  "name": "pdm",
  "version": "0.0.0",
  "scripts": {
    "ng": "ng",
    "start": "ng serve",
    "build": "ng build",
    "watch": "ng build --watch --configuration development",
    "test": "ng test"
  },
  "private": true,
  "dependencies": {
    "@angular/animations": "^14.1.0",
    "@angular/common": "^14.1.0",
    "@angular/compiler": "^14.1.0",
    "@angular/core": "^14.1.0",
    "@angular/forms": "^14.1.0",
    "@angular/platform-browser": "^14.1.0",
    "@angular/platform-browser-dynamic": "^14.1.0",
    "@angular/router": "^14.1.0",
    "rxjs": "~7.5.0",
    "tslib": "^2.3.0",
    "zone.js": "~0.11.4"
  },
  "devDependencies": {
    "@angular-devkit/build-angular": "^14.1.0",
    "@angular/cli": "~14.1.0",
    "@angular/compiler-cli": "^14.1.0",
    "@types/jasmine": "~4.0.0",
    "eslint": "^8.20.0",
    "jasmine-core": "~4.2.0",
    "karma": "~6.4.0",
    "karma-chrome-launcher": "~3.1.0",
    "karma-coverage": "~2.2.0",
    "karma-jasmine": "~5.1.0",
    "karma-jasmine-html-reporter": "~2.0.0",
    "typescript": "~4.7.2"
  }
}
```
```js
/* eslint-disable no-undef */
// Karma configuration file, see link for more information
// https://karma-runner.github.io/1.0/config/configuration-file.html

module.exports = function (config) {
  config.set({
    basePath: '',
    frameworks: ['jasmine', '@angular-devkit/build-angular'],
    plugins: [
      require('karma-jasmine'),
      require('karma-chrome-launcher'),
      require('karma-jasmine-html-reporter'),
      require('karma-coverage'),
      require('@angular-devkit/build-angular/plugins/karma')
    ],
    client: {
      jasmine: {
        // you can add configuration options for Jasmine here
        // the possible options are listed at https://jasmine.github.io/api/edge/Configuration.html
        // for example, you can disable the random execution with `random: false`
        // or set a specific seed with `seed: 4321`
      },
      clearContext: false // leave Jasmine Spec Runner output visible in browser
    },
    jasmineHtmlReporter: {
      suppressAll: true // removes the duplicated traces
    },
    coverageReporter: {
      dir: require('path').join(__dirname, './coverage/pdm'),
      subdir: '.',
      reporters: [
        { type: 'html' },
        { type: 'text-summary' }
      ]
    },
    reporters: ['progress', 'kjhtml'],
    port: 9876,
    colors: true,
    logLevel: config.LOG_INFO,
    autoWatch: true,
    browsers: ['Chrome'],
    singleRun: false,
    restartOnFileChange: true
  });
};
```

**Comment by lucono** [comment]
@speedyscooter This seems related and might be a duplicate of issue #37. Try the fix in the `beta` branch by downloading the [`.vsix`](https://github.com/lucono/karma-test-explorer/suites/7537387612/artifacts/310724175) file from that CI build to confirm if it resolves the issue for you.

**Comment by speedyscooter** [comment]
Hi, I tried to pull down the code and followed some of the directions but I got stuck at the generation of the vsix.
I tried pulling down the above and loading it but it appears that it is the same as the version.
Released on
9/28/2021, 21:37:06
Last updated
7/24/2022, 13:09:20
Identifier
lucono.karma-test-explorer

Note, I tried to build this locally and had the same issue.  Are the dates set somewhere else?  Also noted that it referred to the master branch in the npm run package which made me concerned that it was picking up the wrong version.

**Comment by lucono** [comment]
@speedyscooter - It's been released as [v0.7.2](https://github.com/lucono/karma-test-explorer/blob/master/CHANGELOG.md#072---jul-27-2022), so you can simply update from the VSCode extensions tab and verify if it resolves the issue.

**Comment by speedyscooter** [comment]
> 
Hi, did not work.  I get the following errors.

Server start failed on port 9876: Error: The '@angular-devkit/build-angular/plugins/karma' karma plugin is meant to be used from within Angular CLI and will not work correctly outside of it.

When I switch back to 0.6.1 it works ok... still has problems every time you run in debug it will not release the server correclty.

**Comment by Randelung** [comment]
Hello, I also get the error mentioned above in the Karma Server output, except with a different port (9976). I'm on 0.7.2 and Angular 14.1.
Not sure if significant, but I'm not running a global `ng`, just a local installation, so I have `npm run ng` in the `angularProcessCommand`.
Happy to provide further details.

**Comment by Randelung** [comment]
This might actually help (From `Karma Test Explorer (workspace)`):
```
[2022-07-28 19:44:22.049] [DEBUG] [ProjectFactory]: Cannot get Angular projects - Angular Json file does not exist: /Users/<user>/AppData/Local/Programs/Microsoft VS Code/D:/Desktop/work/engineering/client/<project>/angular.json
[2022-07-28 19:44:22.049] [DEBUG] [ProjectFactory]: Cannot get Angular projects - Angular CLI Json file does not exist: /Users/<user>/AppData/Local/Programs/Microsoft VS Code/D:/Desktop/work/engineering/client/<project>/.angular-cli.json
[2022-07-28 19:44:22.049] [INFO] [ProjectFactory]: Using project type 'karma' (auto-detected) for workspace folder: D:/Desktop/work/engineering/client/<project>
```

Looks like project type detection doesn't check the correct folder. The D:\ concatenation is definitely wrong.

With a statically configured Angular project:
```
[2022-07-28 19:49:18.943] [DEBUG] [ProjectFactory]: Cannot get Angular projects - Angular Json file does not exist: /Users/<user>/AppData/Local/Programs/Microsoft VS Code/D:/Desktop/work/engineering/client/<project>/angular.json
[2022-07-28 19:49:18.943] [DEBUG] [ProjectFactory]: Cannot get Angular projects - Angular CLI Json file does not exist: /Users/<user>/AppData/Local/Programs/Microsoft VS Code/D:/Desktop/work/engineering/client/<project>/.angular-cli.json
[2022-07-28 19:49:18.943] [WARN] [ProjectFactory]: Project type is configured as angular but no viable angular projects were found for workspace folder 'D:/Desktop/work/engineering/client/<project>'
[2022-07-28 19:49:18.943] [INFO] [ProjectFactory]: Using project type 'karma'  for workspace folder: D:/Desktop/work/engineering/client/<project>
```

**Comment by Randelung** [comment]
Yeah, `posix.resolve` doesn't like Windows, probably. Before:
```
[2022-07-28 20:11:01.896] [INFO] [ProjectFactory]: D:/Desktop/work/engineering/client/<project>
```
After:
```
[2022-07-28 20:11:01.896] [INFO] [ProjectFactory]: /Users/<user>/AppData/Local/Programs/Microsoft VS Code/D:/Desktop/work/engineering/client/<project>/angular.json
```

Using regular `path.resolve` will at least detect the agular project,  but I'm not confident it's that easy or doesn't break Linux compatibility.

**Comment by lucono** [comment]
@speedyscooter Could you also share:
- You project folder structure (Run `tree` in the project root after moving out `node_modules`)
- Your Karma Test Explorer extension settings

**Comment by lucono** [comment]
@Randelung I've released your fix to the marketplace as `v0.7.3`, thanks! @speedyscooter - Update to the new version in VSCode and confirm if any other issues.
